redash/models.py

```diff
@@ -769,9 +769,13 @@ def store_result(cls, org, data_source, query_hash, query, data, run_time, retri
         queries = db.session.query(Query).filter(
             Query.query_hash == query_hash,
             Query.data_source == data_source)
+
         for q in queries:
             q.latest_query_data = query_result
             db.session.add(q)
+            if q.schedule_keep_results > 0:
+
+                q.query_results.append(query_result)
```

Comment:
That's basically to have a consistent api to access the results of a query via the `query.query_results` list?

Reply:
This creates a `QueryResultSet` row.